Added config and documentation for setup of self-hosted azure (#1391)
conda-forge/conda-forge.github.io: src/maintainer/infrastructure.rst

``conda-smithy`` also contains the command line tool that you should use if you rerender manually from the command line (see :ref:`dev_update_rerender`).

> Smithy can be used for more than just CF; it can also be used to set up self-hosted agents for non-CF infrastructures.

Reviewer: I don't think we refer to "CF" in this file, so let's avoid it:
```suggestion
Smithy can be used beyond Conda-Forge's purposes. For example, it can be used to set up self-hosted Azure agents for non-Conda-Forge infrastructures.
```

> Smithy can be used beyond Conda-Forge's purposes. For example, it can be used to set up self-hosted Azure agents for non-Conda-Forge infrastructures.

Reviewer: ```suggestion
Smithy can be used beyond Conda-Forge's purposes. For example, it can be used to `set up self-hosted Azure agents <self-hosted_azure-config>`_ for non-Conda-Forge infrastructures.
```

> Smithy can be used beyond Conda-Forge's purposes. For example, it can be used to `set up self-hosted Azure agents <self-hosted_azure-config>`_ for non-Conda-Forge infrastructures.

Reviewer 2: `conda-forge` channel or Conda-Forge org?

Author: Well i am also confused now let's ask @leofang

> For autoscaling agents, you can also use `Azure virtual machine scale set agents <https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops>`_ , 
> which are less expensive to run than permanently active agents. 

Reviewer: Sorry, but I am now a little concerned about bringing this up, as its connection to self-hosted agents wasn't obvious at first glance. How about using parenthesis to make it look like a drop-by mention?
```suggestion
(You could also consider using `Azure virtual machine scale set agents <https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/scale-set-agents?view=azure-devops>`_,
which could be less expensive to run than permanently active agents.)
```
